php-frontend/src/main/java/org/sonar/php/tree/impl/expression/AnonymousClassTreeImpl.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.tree.impl.expression;
+
+import com.google.common.collect.Iterators;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.sonar.php.tree.impl.PHPTree;
+import org.sonar.php.tree.impl.SeparatedListImpl;
+import org.sonar.plugins.php.api.tree.SeparatedList;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.ClassMemberTree;
+import org.sonar.plugins.php.api.tree.declaration.MethodDeclarationTree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.AnonymousClassTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.visitors.VisitorCheck;
+
+public class AnonymousClassTreeImpl extends PHPTree implements AnonymousClassTree {
+
+  private static final Kind KIND = Kind.ANONYMOUS_CLASS;
+
+  private final SyntaxToken classToken;
+  private final SyntaxToken openParenthesisToken;
+  private final SeparatedList<ExpressionTree> arguments;
+  private final SyntaxToken closeParenthesisToken;
+  private final SyntaxToken extendsToken;
+  private final NamespaceNameTree superClass;
+  private final SyntaxToken implementsToken;
+  private final SeparatedListImpl<NamespaceNameTree> superInterfaces;
+  private final SyntaxToken openCurlyBraceToken;
+  private final List<ClassMemberTree> members;
+  private final SyntaxToken closeCurlyBraceToken;
+
+  public AnonymousClassTreeImpl(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 11 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)